sinonjs/referee
# Upgrade @sinonjs/eslint-config

This PR upgrades ~`eslint-config-sinon`~ `@sinonjs/eslint-config` and fixes all uncovered lint violations.

#### Purpose

Adopt the latest `@sinonjs/eslint-config` in order to allow modern JavaScript syntax and ease the maintenance burden of keeping all the development tools up to date.

#### Background

https://hackmd.io/@mroderick/sinon-rfc-drop-legacy-browsers

#### Solution

* Add latest version of `@sinonjs/eslint-config` as `devDependency`
* Fix all lint violations
* Configure pre-commit hook and CI to ignore lint warnings. This allows contributions before we add all the missing JSDoc comments. We can still see all warnings by running `npm run lint`
* Uninstall no longer needed linting dependencies

#### TO DO

- [ ] Install `@sinonjs/eslint-config`
- [ ] Remove `eslint`
- [ ] Remove `eslint-config-prettier`
- [ ] Remove `eslint-plugin-ie11`
- [ ] Remove `eslint-plugin-mocha`
- [ ] Remove `eslint-plugin-prettier`

#### How to verify - mandatory

1. Check out this branch
2. `npm install`
3. DRAFT - until https://github.com/sinonjs/eslint-config-sinon/pull/7 is merged and published
    1. Check out https://github.com/sinonjs/eslint-config-sinon/pull/7 somewhere
    2. In that folder
        1. `npm link`
    3. In the `referee` repository
        1. `npm link @sinonjs/eslint-config`
        4. `npm install eslint-plugin-compat`
        5. `npm install eslint-plugin-jsdoc` 
        6. `npm install eslint-plugin-mocha`
3. `npm run lint`
4. Observe that the only warnings are `jsdoc/*`
5. Try creating an `async` function without an `await` statement
6. Observe that the linter complains about it
7. Add an `await` statement
8. Observe that the linter allows `async` functions

#### Checklist for author
